extensions/amp-timeago/0.2/timeago.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {useResourcesNotify} from '../../../src/preact/utils';
+import {timeago} from '../../../third_party/timeagojs/timeago';
+import { createElement } from 'preact';
+
+/**
+ * @param {!JsonObject} props
+ * @return {Preact.Renderable}
+ */
+export function Timeago(props) {
+  let content;
+  if (props['cutoff']) {
```

Comment:
are you using `props['cutoff']` instead of `props.cutoff` because of obfuscation here?

Reply:
See the discussion at https://github.com/ampproject/amphtml/pull/25969#discussion_r365483372. We might be able to use an extern'd typedef, but for now, it's necessary to use computed-string form.

Reply:
We should invest quickly in automatic externs for these components. This pattern is not idiomatic.

Reply:
Strongly agree. But wondering if we should simply put the React code into separate folders (with tests) and run a Babel plugin to make them compilable into AMP.

Reply:
That's also a good option. Would make the code far more parseable.